src/runtime/recipe/recipe-resolver.ts

```diff
@@ -57,11 +53,11 @@ export class ResolveWalker extends RecipeWalker {
       const counts = RecipeUtil.directionCounts(handle);
       switch (handle.fate) {
         case 'use':
-          mappable = arc.findStoresByType(handle.type, {tags: handle.tags});
+          mappable = storeHolder.findStoresByType(handle.type, {tags: handle.tags});
           break;
         case 'map':
         case 'copy':
-          mappable = arc.context.findStoresByType(handle.type, {tags: handle.tags, subtype: true});
+          mappable = storeHolder.context.findStoresByType(handle.type, {tags: handle.tags, subtype: true});
```

Comment:
Why this is not `storeHolder.findStoresByType`, i.e. without the context in between?

Reply:
I am wary about modifying the recipe resolution algorithm. As I understand it, the context is here because in resolution, we want to check for stores in the underlying manifest or in the arc depending on how the fate of the handle. While it doesn't make sense in terms of the new interface, it does make sense in terms of concrete class definitions (i.e. `Arc` and `Manifest`).